replicators/src/mysql_connector/connector.rs

```diff
@@ -128,8 +132,31 @@ impl MySqlBinlogConnector {
         };
 
         connector.register_as_replica().await?;
-        connector.request_binlog().await?;
-
+        let binlog_request = connector.request_binlog().await;
+        match binlog_request {
+            Ok(()) => (),
+            Err(mysql_async::Error::Server(ref err))
+                if mysql_srv::ErrorKind::from(err.code)
+                    == mysql_srv::ErrorKind::ER_MASTER_FATAL_ERROR_READING_BINLOG =>
+            {
+                // Requested binlog is not available, we need to re-snapshot
+                error!("Failed to request binlog: {}", err);
+                return Err(ReadySetError::FullResnapshotNeeded);
+            }
+            Err(mysql_async::Error::Other(ref err)) => {
+                error!("Failed to request binlog: {}", err);
+                if err.downcast_ref::<ReadySetError>() == Some(&ReadySetError::FullResnapshotNeeded)
+                {
+                    return Err(ReadySetError::FullResnapshotNeeded);
+                }
+            }
+            Err(err) => {
+                return Err(ReadySetError::Internal(format!(
+                    "Failed to request binlog: {}",
+                    err
+                )));
```

Comment:
the `internal!` macro does this but also puts line number info in dev mode:
```suggestion
                internal!("Failed to request binlog: {err}")
```

Reply:
Done